source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m

```diff
@@ -462,6 +463,19 @@ - (void)didChangeViewLayout:(CGRect)oldFrame newFrame:(CGRect)newFrame
     [self.scrView scrollRectToVisible:newFrame animated:YES];
 }
 
+- (void)didChangeViewLayout:(CGRect)oldFrame newFrame:(CGRect)newFrame properties:(NSDictionary *)properties
+{
+    NSString *actiontype = (NSString *)properties[ACRAggregateTargetActionType];
+    if ([actiontype isEqualToString:ACRAggregateTargetSubmitAction]) {
```

Comment:
why does this only target submit actions?

Reply:
This is for input layout update which is handled by a submit action type. For example, a view needs to be scrolled, so users can see what's invalid. The scrolling requires a superview to change its content offset. Also, if the superview is a UITableView, it needs to refresh its layout when hiddle error message is unhidden.  I intended only for a custom action, but I liked behavior, so I added this key, ACRAggregateTragetSubmitAction.